src/lib.rs

```diff
@@ -2060,6 +2060,68 @@ impl Connection {
     /// # Ok::<(), quiche::Error>(())
     /// ```
     pub fn send(&mut self, out: &mut [u8]) -> Result<usize> {
+        if out.is_empty() {
+            return Err(Error::BufferTooShort);
+        }
+
+        let mut has_initial = false;
+
+        let mut done = 0;
+
+        // Limit output packet size to respect the sender and receiver's
+        // maximum UDP payload size limit.
+        let mut left = cmp::min(out.len(), self.max_send_udp_payload_len());
+
+        // Limit data sent by the server based on the amount of data received
+        // from the client before its address is validated.
+        if !self.verified_peer_address && self.is_server {
+            left = cmp::min(left, self.max_send_bytes);
+        }
+
+        // Generate coalesced packets.
+        while left > 0 {
+            let (ty, written) =
+                match self.send_single(&mut out[done..done + left]) {
+                    Ok(v) => v,
+
+                    Err(Error::BufferTooShort) | Err(Error::Done) => break,
+
+                    Err(e) => return Err(e),
+                };
+
+            done += written;
+            left -= written;
+
+            match ty {
+                packet::Type::Initial => has_initial = true,
+
+                // No more packets can be coalesced after a 1-RTT.
+                packet::Type::Short => break,
+
+                _ => (),
+            };
+        }
+
+        if done == 0 {
+            return Err(Error::Done);
+        }
+
+        // Pad UDP datagram if it contains a QUIC Initial packet.
+        if has_initial && left > 0 && done < MIN_CLIENT_INITIAL_LEN {
+            let pad_len = cmp::min(left, MIN_CLIENT_INITIAL_LEN - done);
+
+            // Fill padding area with null bytes, to avoid leaking information
+            // in case the application reuses the packet buffer.
+            let pad_ptr = (&mut out[done..done + pad_len]).as_mut_ptr();
+            unsafe { std::ptr::write_bytes(pad_ptr, 0, pad_len) };
```

Comment:
```suggestion
            out[done..done + pad_len].fill(0);
```
Looks like it can be simplified and safe as above, rust 1.50 and later (CI use already 1.50)
https://doc.rust-lang.org/std/primitive.slice.html#method.fill

Reply:
Updated, thanks.